services/subscribers/subscriber.go

```diff
@@ -0,0 +1,203 @@
+package subscribers
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/jinzhu/gorm"
+
+	"github.com/mailbadger/app/entities"
+	"github.com/mailbadger/app/storage"
+)
+
+type Service interface {
+	ImportSubscribersFromFile(ctx context.Context, filename string, userID int64, segments []entities.Segment) error
+	RemoveSubscribersFromFile(ctx context.Context, filename string, userID int64) error
+	DeactivateSubscriber(ctx context.Context, userID int64, email string) error
+}
+
+type service struct {
+	client s3iface.S3API
+	db     storage.Storage
+}
+
+var (
+	ErrInvalidColumnsNum = errors.New("invalid number of columns")
+	ErrInvalidFormat     = errors.New("csv file not formatted properly")
+)
+
+func New(client s3iface.S3API, db storage.Storage) *service {
+	return &service{client, db}
+}
+
+func (s *service) ImportSubscribersFromFile(
+	ctx context.Context,
+	filename string,
+	userID int64,
+	segments []entities.Segment,
+) (err error) {
+	res, err := s.client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(os.Getenv("FILES_BUCKET")),
+		Key:    aws.String(fmt.Sprintf("subscribers/import/%d/%s", userID, filename)),
+	})
+	if err != nil {
+		return fmt.Errorf("importer: get object: %w", err)
+	}
+	defer func() {
+		if cerr := res.Body.Close(); cerr != nil {
+			err = cerr
+		}
+	}()
+
+	reader := csv.NewReader(res.Body)
+	header, err := reader.Read()
+	if err != nil {
+		if err == io.EOF {
+			return fmt.Errorf("importer: empty file '%s': %w", filename, err)
+		}
+
+		return fmt.Errorf("importer: read header: %w", err)
+	}
+
+	if len(header) < 2 {
+		return ErrInvalidColumnsNum
+	}
+
+	if strings.ToLower(header[0]) != "email" || strings.ToLower(header[1]) != "name" {
+		return ErrInvalidFormat
+	}
+
+	for {
+		line, err := reader.Read()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return fmt.Errorf("importer: read line: %w", err)
+		}
+		if len(line) < 2 {
+			continue
+		}
+		email := strings.TrimSpace(line[0])
+		name := strings.TrimSpace(line[1])
+
+		_, err = storage.GetSubscriberByEmail(ctx, email, userID)
```

Comment:
Since we'll use the storage as a dependency we can refactor this.
```suggestion
		_, err = s.db.GetSubscriberByEmail(email, userID)
```